ui/perfherder/alerts/Pagination.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';
+
+import withValidation from '../Validation';
+
+class PaginationGroup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      page: this.props.page,
+      count: 0,
+    };
+  }
+
+  navigatePage = page => {
+    this.setState({ page }, this.props.fetchData);
+    this.props.updateParams({ page });
+  };
+
+  render() {
+    const { pageNums } = this.props;
+    const { page, count } = this.state;
+    return (
+      <Pagination aria-label={`Page ${page}`}>
+        {page > 1 && (
+          <PaginationItem>
+            <PaginationLink
+              className="text-info"
+              previous
+              onClick={() => this.navigatePage(page - 1)}
+            />
+          </PaginationItem>
+        )}
+        {pageNums.map(num => (
+          <PaginationItem
+            key={num}
+            active={num === page}
+            className="text-info pagination-active"
+          >
+            <PaginationLink
+              className="text-info"
+              onClick={() => this.navigatePage(num)}
+            >
+              {num}
+            </PaginationLink>
+          </PaginationItem>
+        ))}
+        {page < count && (
+          <PaginationItem>
+            <PaginationLink
+              className="text-info"
+              next
+              onClick={() => this.navigatePage(page + 1)}
+            />
+          </PaginationItem>
+        )}
+      </Pagination>
+    );
+  }
+}
+
+PaginationGroup.propTypes = {
+  pageNums: PropTypes.array.isRequired,
+  page: PropTypes.number,
+  fetchData: PropTypes.func.isRequired,
+};
+
+PaginationGroup.defaultProps = {
+  page: 1,
+};
+
+export default withValidation(
```

Comment:
You don't need to use `withValidation` here; it's a higher order component that passes validated query param props to whichever component you wrap it in but isn't relevant to this component.

Reply:
That was also causing an error while testing out this pr locally. Always a good idea to test your changes thoroughly on your local environment, even seemingly small changes :)